src/lib/core/components/VariableLink.tsx

```diff
@@ -1,32 +1,66 @@
 /*
  * Link to the page for a variable
  */
-import { useMemo } from 'react';
+import { forwardRef, Ref } from 'react';
 import { Link, LinkProps } from 'react-router-dom';
-import { useMakeVariableLink } from '../hooks/workspace';
+
+type VariableValue = {
+  entityId?: string;
+  variableId?: string;
+};
+
+export type VariableLinkConfig =
+  | {
+      type: 'link';
+      makeVariableLink: (value?: VariableValue) => string;
+    }
+  | {
+      type: 'button';
+      onClick: (value?: VariableValue) => void;
+    };
 
 export interface Props<S = unknown> extends Omit<LinkProps<S>, 'to'> {
   entityId?: string;
   variableId?: string;
+  linkConfig: VariableLinkConfig;
 }
 
-export function VariableLink(props: Props) {
-  const { entityId, variableId, ...rest } = props;
-  const makeVariableLink = useMakeVariableLink();
-  const variableLink = useMemo(
-    () =>
-      makeVariableLink({
-        entityId,
-        variableId,
-      }),
-    [makeVariableLink, entityId, variableId]
-  );
+export const VariableLink = forwardRef(
+  (props: Props, ref: Ref<HTMLAnchorElement>) => {
+    const { entityId, variableId, linkConfig, ...rest } = props;
+    const value = { entityId, variableId };
 
-  return (
-    <Link
-      replace
-      {...rest}
-      to={{ pathname: variableLink, state: { scrollToTop: false } }}
-    />
-  );
-}
+    return linkConfig.type === 'link' ? (
+      <Link
+        replace
+        {...rest}
+        to={{
+          pathname: linkConfig.makeVariableLink(value),
+          state: { scrollToTop: false },
+        }}
+      />
+    ) : (
+      // Typically, we would just use a <button> for this case,
+      // but this needs to work in an SVG fragment, which does
+      // not support <button>, so this is a compromise
+      // eslint-disable-next-line jsx-a11y/anchor-has-content
+      <a
+        ref={ref}
+        role="button"
+        aria-pressed="false"
```

Comment:
**question**: will `aria-pressed` update on press?

Reply:
Oh, yeah. I'll look at this in more detail soon. I might be missing some other aria attributes...

Reply:
I think the solution is to remove `aria-pressed`. That's more useful for indicating toggle state, which is not needed here.

Reply:
I'm now really wanting to use a `<button>`. I will explore using `<foreignObject>` in EntityDiagram.

Reply:
`<foreignObject>` really doesn't seem viable. I'll stick with the current approach. Hopefully is meets a11y requirements. @adnauseum do you know of a good tool for checking that?

Reply:
Well, if we're ever in a situation were we have like an ADA audit: [Assistive Labs](https://assistivlabs.com/pricing) is a good choice, but that is most definitely not free 😄 . The free options are not great. I use [this Chrome Extension](https://chrome.google.com/webstore/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn?hl=en) to make sure things are useable. I think it's process of gradual enhancement anyways